### DISCUSSION TITLE
赞！v2.60 终于解决了 el-checkbox el-radio el-selection 产出值不明确的问题， label 和 value 的问题

### DISCUSSION THREAD

**Discussion by KyleBing** [element-plus/element-plus#16050] [Show and tell]
昨天想反馈这个关于 label 和 value 的问题的时候，发现新版本已经修正这个问题了。

### 一、ElementUI 旧版关于 label 和 value 的问题
从之前 element-ui 用过来的，之前就一直很烦 element 关于 radio checkbox 取值不明的问题：
label 应该是显示的名，value 应为实际产出的值

```html
<el-radio label="选项1" :value="1"/>
```
但，在旧版中不是这样的，每次用到都很烦，它产出的值竟然是 label，简直太傻了。
比如这样一个例子，在 v2.42 版本中的产出值是这样。

> 在线例子： [v2.42 例子](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3Qgc2VsZWN0VmFsdWUgPSByZWYoMSlcbmNvbnN0IGNoZWNrYm94VmFsdWUgPSByZWYoW10pXG5jb25zdCByYWRpb1ZhbHVlID0gcmVmKDEpXG48L3NjcmlwdD5cblxuXG48dGVtcGxhdGU+XG48aDM+IEZvcm0gSXRlbSBUZXN0PC9oMz5cblxuPGVsLXNlbGVjdCB2LW1vZGVsPVwic2VsZWN0VmFsdWVcIj5cbiAgPGVsLW9wdGlvbiBsYWJlbD1cIkxhYmVsMVwiIDp2YWx1ZT1cIjFcIj48L2VsLW9wdGlvbj5cbiAgPGVsLW9wdGlvbiBsYWJlbD1cIkxhYmVsMlwiIDp2YWx1ZT1cIjJcIj48L2VsLW9wdGlvbj5cbjwvZWwtc2VsZWN0PlxuXG48ZWwtY2hlY2tib3gtZ3JvdXAgdi1tb2RlbD1cImNoZWNrYm94VmFsdWVcIj5cbiAgPGVsLWNoZWNrYm94IGxhYmVsPVwiY2hlY2tib3gxXCIgOnZhbHVlPVwiMVwiPjwvZWwtY2hlY2tib3g+XG4gIDxlbC1jaGVja2JveCBsYWJlbD1cImNoZWNrYm94MlwiIDp2YWx1ZT1cIjJcIj48L2VsLWNoZWNrYm94PlxuPC9lbC1jaGVja2JveC1ncm91cD5cblxuPGVsLXJhZGlvLWdyb3VwIHYtbW9kZWw9XCJyYWRpb1ZhbHVlXCI+XG4gIDxlbC1yYWRpbyBsYWJlbD1cImNoZWNrYm94MVwiIDp2YWx1ZT1cIjFcIj48L2VsLXJhZGlvPlxuICA8ZWwtcmFkaW8gbGFiZWw9XCJjaGVja2JveDJcIiA6dmFsdWU9XCIyXCI+PC9lbC1yYWRpbz5cbjwvZWwtcmFkaW8tZ3JvdXA+XG5cbjxocj5cblxuPHAgY2xhc3M9XCJkaXNwbGF5XCI+c2VsZWN0IHZhbHVlID0ge3tzZWxlY3RWYWx1ZX19PC9wPlxuPHAgY2xhc3M9XCJkaXNwbGF5XCI+Y2hlY2tib3ggdmFsdWUgPSB7e2NoZWNrYm94VmFsdWV9fTwvcD5cbjxwIGNsYXNzPVwiZGlzcGxheVwiPnJhZGlvIHZhbHVlID0ge3tyYWRpb1ZhbHVlfX08L3A+XG5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)  记得将版本切到 2.42 再试

```html
<template>
<h3> Form Item Test</h3>

<el-select v-model="selectValue">
  <el-option label="Label1" :value="1"></el-option>
  <el-option label="Label2" :value="2"></el-option>
</el-select>

<el-checkbox-group v-model="checkboxValue">
  <el-checkbox label="checkbox1" :value="1"></el-checkbox>
  <el-checkbox label="checkbox2" :value="2"></el-checkbox>
</el-checkbox-group>

<el-radio-group v-model="radioValue">
  <el-radio label="checkbox1" :value="1"></el-radio>
  <el-radio label="checkbox2" :value="2"></el-radio>
</el-radio-group>

<hr>

<p class="display">select value = {{selectValue}}</p>
<p class="display">checkbox value = {{checkboxValue}}</p>
<p class="display">radio value = {{radioValue}}</p>

</template>
```

<img width="506" alt="Screenshot 2024-03-06 at 09 15 58" src="https://github.com/element-plus/element-plus/assets/12215982/dd52fcb3-b671-4875-9bf7-81e9546856f9">

### 二、v2.60 已经修正了 label value 取值不明确的问题

而换成 v2.60 之后就正常了，取的就是 value 的值。

> 在线例子： [v2.60 例子](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuY29uc3Qgc2VsZWN0VmFsdWUgPSByZWYoMSlcbmNvbnN0IGNoZWNrYm94VmFsdWUgPSByZWYoW10pXG5jb25zdCByYWRpb1ZhbHVlID0gcmVmKDEpXG48L3NjcmlwdD5cblxuXG48dGVtcGxhdGU+XG48aDM+IEZvcm0gSXRlbSBUZXN0PC9oMz5cblxuPGVsLXNlbGVjdCB2LW1vZGVsPVwic2VsZWN0VmFsdWVcIj5cbiAgPGVsLW9wdGlvbiBsYWJlbD1cIkxhYmVsMVwiIDp2YWx1ZT1cIjFcIj48L2VsLW9wdGlvbj5cbiAgPGVsLW9wdGlvbiBsYWJlbD1cIkxhYmVsMlwiIDp2YWx1ZT1cIjJcIj48L2VsLW9wdGlvbj5cbjwvZWwtc2VsZWN0PlxuXG48ZWwtY2hlY2tib3gtZ3JvdXAgdi1tb2RlbD1cImNoZWNrYm94VmFsdWVcIj5cbiAgPGVsLWNoZWNrYm94IGxhYmVsPVwiY2hlY2tib3gxXCIgOnZhbHVlPVwiMVwiPjwvZWwtY2hlY2tib3g+XG4gIDxlbC1jaGVja2JveCBsYWJlbD1cImNoZWNrYm94MlwiIDp2YWx1ZT1cIjJcIj48L2VsLWNoZWNrYm94PlxuPC9lbC1jaGVja2JveC1ncm91cD5cblxuPGVsLXJhZGlvLWdyb3VwIHYtbW9kZWw9XCJyYWRpb1ZhbHVlXCI+XG4gIDxlbC1yYWRpbyBsYWJlbD1cImNoZWNrYm94MVwiIDp2YWx1ZT1cIjFcIj48L2VsLXJhZGlvPlxuICA8ZWwtcmFkaW8gbGFiZWw9XCJjaGVja2JveDJcIiA6dmFsdWU9XCIyXCI+PC9lbC1yYWRpbz5cbjwvZWwtcmFkaW8tZ3JvdXA+XG5cbjxocj5cblxuPHAgY2xhc3M9XCJkaXNwbGF5XCI+c2VsZWN0IHZhbHVlID0ge3tzZWxlY3RWYWx1ZX19PC9wPlxuPHAgY2xhc3M9XCJkaXNwbGF5XCI+Y2hlY2tib3ggdmFsdWUgPSB7e2NoZWNrYm94VmFsdWV9fTwvcD5cbjxwIGNsYXNzPVwiZGlzcGxheVwiPnJhZGlvIHZhbHVlID0ge3tyYWRpb1ZhbHVlfX08L3A+XG5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319) 记得将版本切到 2.60 再试

<img width="628" alt="Screenshot 2024-03-06 at 09 18 30" src="https://github.com/element-plus/element-plus/assets/12215982/9de8ca46-4adf-4658-82f1-5bb2d425a8bc">

**Comment by kooriookami** [comment]
Vue2 中 value 作为 v-model 的绑定值有特殊含义，而 Vue3 中 v-model 的值是 modelValue，所以 value 可以用来做其他属性。

**Comment by elarbor** [comment]
还是[elementUI-vue3](https://element-plus.org/)用的惯，别的组件库我都用不惯😂
